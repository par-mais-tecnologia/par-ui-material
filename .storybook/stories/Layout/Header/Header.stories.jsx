import { storiesOf } from '@storybook/react'
import React from 'react'
import withTests from '../../withTests'
import { 
  Header, 
  BioFinanceiraTheme, 
  MuiThemeProvider, 
  SeeTheme 
} from '../../../../src'
import Typography from '@material-ui/core/Typography/Typography'
import style from '../../style'

const HeaderStory = () => (
  <MuiThemeProvider theme={SeeTheme}>
    <div style={{...style.section,  gridTemplateColumns: 'repeat(auto-fill)', padding: '2rem'}}>
      <div style={style.sectionItemWrapper}>
        <Typography variant='title'>
          How to use?
        </Typography>
        <Typography variant='body2'>
          To use it you just need to wrap the content with the component and enjoy it!
        </Typography>
        <div style={style.sectionItem}>
          <div style={{...style.sectionContent, background: 'white', width: '100%'}}>
          <Header
            classes='justify-center'
            boxShadow={'0px -3px 20px 0px'}
            backgroundImage={BioFinanceiraTheme.styles.colors.seeGradient}
            height='70px' margin='16px'
            justifyContent='center'>
            <Typography align='center' style={BioFinanceiraTheme.styles.fonts.mainTitle}>
              <b>BIO</b> <span style={{fontWeight: 100}}>FINANCEIRA</span>
            </Typography>
          </Header>
          </div>
          <Typography variant='body1'>
            &lt;<span style={style.tag}>Header</span>&gt; ...
            &lt;/<span style={style.tag}>Header</span>&gt;
          </Typography>
        </div>
      </div>
    </div>
  </MuiThemeProvider>
)

storiesOf('Layout / Header', module)
  .addDecorator(withTests('Header'))
  .add('Bio Financeira', () => {
    return <HeaderStory />
  })
