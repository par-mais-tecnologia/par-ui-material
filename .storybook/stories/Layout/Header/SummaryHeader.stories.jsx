import { storiesOf } from '@storybook/react'
import React, { PureComponent } from 'react'
import 'tachyons/css/tachyons.min.css'
import withTests from '../../withTests'
import { 
  SummaryHeader, 
  SeeTheme, 
  IconButton, 
  RepeatIcon, 
  Typography 
} from '../../../../src'
import '../../../../src/Core/styles/transitions.css'
import { MuiThemeProvider } from '@material-ui/core'
import style from '../../style'

let seeThemeOverride = SeeTheme;
seeThemeOverride.overrides.MuiIconButton = {
  root: {
    '&:hover': {
      backgroundColor: 'none'
    },
    '&:link': {
      backgroundColor: 'none'
    }
  }
},seeThemeOverride.overrides.MuiTouchRipple = {
    root: {
      color: 'transparent'
    }
}


class SummaryHeaderStory extends PureComponent {

  state = {
    value : 12.3,
    liquidGross: true,
  }

  numberFormat(number) {
    if (isNaN(number) || number === null) {
      return defaultValue
    }

    let numberValue = Number(number)

    let salary = this.state.liquidGross ? numberValue - (numberValue * 0.11) : numberValue

    const value = 'R$' + salary.toLocaleString('pt-BR', {minimumFractionDigits:2, maximumFractionDigits:2});

    return value.replace(/\./g, ',');
  }

  getHelperText() {
    if(this.state.liquidGross) {
      return 'É o valor líquido total aplicado'
    } else {
      return 'É o valor bruto total aplicado'
    }
  }

  handleChangeTypes = () => {
    this.setState({liquidGross: !this.state.liquidGross})
  }

  render () {
    const { value } = this.state

    return (
      <MuiThemeProvider theme={seeThemeOverride}>
        <div style={{...style.section,  gridTemplateColumns: 'repeat(auto-fill)', padding: '2rem'}}>
          <div style={style.sectionItemWrapper}>
            <div>
              <Typography variant='title'>
                How to use?
              </Typography>
              <Typography variant='body2'>
                The style of <span style={style.tag}>SummaryHeader </span> is part of 
                <span style={style.tag}>SeeTheme </span> theme, to use it you must import the files
                into your project.
                This component accept a prop named as <span style={style.attr}>headerProps</span>, which is an "css" object,
                you should use with the intend to style the component.
              </Typography>
            </div>
            <div style={style.sectionItem}>
              <div style={{...style.sectionContent, width: '100%', background: 'white'}}>
                <SummaryHeader
                  classes='justify-center'
                  headerProps={{
                    boxShadow:'0px -3px 20px 0px',
                    backgroundImage:SeeTheme.styles.colors.seeGradient,
                    margin:'16px 0',
                    justifyContent:'center'
                  }}>

                  <div className='flex justify-center items-center flex-column'>
                    <div>
                      <Typography
                        align='center'
                        style={{...SeeTheme.styles.fonts.headline, marginTop:'117px', marginBottom:'0px'}}>
                        INVESTIMENTOS TOTAIS
                      </Typography>
                    </div>

                    <div className='flex flex-row'>
                          <div className='left'>
                            <p style={{marginTop:'18px', color:'white', fontSize:'24px',fontFamily:'Roboto Light', marginBottom:'0' }}
                              className={this.state.liquidGross? 'frame1' : 'frame2'}>
                              {this.numberFormat(value)}
                            </p>
                          </div>

                      <div className='right'>
                        <IconButton style={{ padding: '12px 0px 12px 12px' }} onClick={() => this.handleChangeTypes()}>
                          <RepeatIcon style={{color:'white', fontSize:'24px', marginTop:'6px'}} />
                        </IconButton>
                      </div>
                    </div>
                    <div>
                      <Typography align='center'
                        style={{...SeeTheme.styles.fonts.subtitle1, marginBottom:'117px', marginTop:'0', animationFillMode: 'backwards'}}
                        className={this.state.liquidGross? 'frame1' : 'frame2'}>
                          {this.getHelperText()}
                      </Typography>
                    </div>
                  </div>

                </SummaryHeader>
              </div>
              <Typography variant='body1'>
                &lt;
                    <span style={style.tag}>SummaryHeader </span>
                    <span style={style.attr}>headerProps</span>=
                    <span style={style.value}>
                    &#123;
                      &#123; 
                        boxShadow:'0px -3px 20px 0px',
                        backgroundImage:SeeTheme.styles.colors.seeGradient,
                        margin:'16px 0',
                        justifyContent:'center'
                      &#125;
                    &#125;
                    </span>
                &gt;...
              </Typography>
            </div>
            <div>
              <Typography variant='overline' style={style.details}>
                Props
              </Typography>
              <Typography variant='caption'>
                headerProps: PropTypes.object,
                <br/>
                children: PropTypes.node
              </Typography>
            </div>
          </div>
        </div>
      </MuiThemeProvider>
    )
  }
}

storiesOf('Layout / Header', module)
  .addDecorator(withTests('SummaryHeader'))
  .add('SummaryHeader', () =>  <SummaryHeaderStory />)
