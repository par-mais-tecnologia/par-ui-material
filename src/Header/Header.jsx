import React, { PureComponent } from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'

class Header extends PureComponent {
  getStyle (props) {
    return {
      height: props.height,
      backgroundImage: props.backgroundImage,
      boxShadow: props.boxShadow
    }
  }

  render () {
    return (
      <Grid container alignItems='center' style={this.getStyle(this.props)}>
        <Typography align='center' variant='title'>
          {this.props.children}
        </Typography>
      </Grid>
    )
  }
}

Header.defaultProps = {
  height: '70px',
  backgroundImage: 'linear-gradient(26deg, rgb(52, 122, 124), rgb(138, 202, 199))',
  boxShadow: '0px -3px 20px 0px'
}

export default Header
