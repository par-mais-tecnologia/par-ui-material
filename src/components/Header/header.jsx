import {PureComponent} from "react";
import React from "react";
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

export default class Header extends PureComponent {

    getStyle(props) {
        return {
            height: props.height,
        }
    }

    render () {

        const { backgroundColor } = this.props;

        return (
            <Grid container spacing={24} className={backgroundColor} style={this.getStyle(this.props)}>
                <Grid item xs={12} className='centralize'>
                    <Typography align='center' variant='title'>
                        {this.props.children}
                    </Typography>
                </Grid>
            </Grid>
        )
    }
}
