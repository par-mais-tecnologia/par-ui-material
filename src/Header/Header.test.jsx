import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Header from './'

configure({ adapter: new Adapter() })

export const test = describe('Header', () => {
  it('shallow Header', () => {
    const wrapper = shallow(
      <Header />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow Header match Snapshot', () => {
    const wrapper = shallow(
      <Header />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
