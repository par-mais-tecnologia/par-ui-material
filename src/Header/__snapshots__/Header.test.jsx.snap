// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header shallow Header match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    backgroundImage="linear-gradient(26deg, rgb(52, 122, 124), rgb(138, 202, 199))"
    boxShadow="0px -3px 20px 0px"
    height="70px"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": undefined,
      "className": "undefined top-0",
      "container": true,
      "style": Object {
        "backgroundImage": "linear-gradient(26deg, rgb(52, 122, 124), rgb(138, 202, 199))",
        "boxShadow": "0px -3px 20px 0px",
        "height": "70px",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": undefined,
        "className": "undefined top-0",
        "container": true,
        "style": Object {
          "backgroundImage": "linear-gradient(26deg, rgb(52, 122, 124), rgb(138, 202, 199))",
          "boxShadow": "0px -3px 20px 0px",
          "height": "70px",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
